### PR TITLE
README.md - better explanation and marketing

### PR DESCRIPTION
**Pull request by sstok**
Explain in a few words what RollerworksSearch is and why its "so great".

/cc @cordoval

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,30 +9,49 @@ README
 What is RollerworksSearch?
 --------------------------
 
-RollerworksSearch provides you with a light yet powerful search system
-to easily integrate into your PHP application.
+RollerworksSearch provides you with a powerful search system
+for your PHP application.
```
Comment (sstok) [inline README.md]: application should be plural

File: README.md [hunk 2]
```diff
@@ -10,30 +9,49 @@ README
...
+whether you use an SQL database system or using a lucene search back-end like
+ElasticSearch or Apache Solr.
 
 ## Input processors
 
 Input processing is provided for the most common formats.
-
 Including a special format called FilterQuery which provides
-a user-friendly syntax for creating simple and complex conditions.
+a user-friendly syntax for creating any type condition.
```
Comment (sstok) [inline README.md]: any type **of**

File: README.md [hunk 3]
```diff
@@ -10,30 +9,49 @@ README
...
 
 The system has a modular design and can work with any PHP framework,
 user locale, data format or storage system.
 
+Whether you want to simply search for users in your SQL database or want to
+provide a powerful search system for searching products using an ElasticSearch
+back-end, everything is possible.
+
+Say your customer data is stored in the "customer" table while the "invoice"
+data is stored in it's own table. And invoice rows also have there own table.
```
Comment (cordoval) [inline README.md]: invoice rows?

File: README.md [hunk 4]
```diff
@@ -10,30 +9,49 @@ README
...
+RollerworksSearch provides you with most of the features you would expect
+from a search system. Including input processing, condition optimizing and
+transforming user input to a normalized data-format. And everything can be
+extended for your own use-cases and needs.
 
-A SearchConditions is build-up of fields and condition-groups.
-Each field can hold any type of value including ranges, comparisons
-and pattern matchers (starts/ends with contains and basic regex).
+Search conditions can be as simple or complex as you need them to be.
+Including grouping and nesting for the best possible result.
```
Comment (cordoval) [inline README.md]: this talks about many things but performance, i wonder if performance is impacted and how

File: README.md [hunk 5]
```diff
@@ -10,30 +9,49 @@ README
...
 
+Whether you want to simply search for users in your SQL database or want to
+provide a powerful search system for searching products using an ElasticSearch
+back-end, everything is possible.
+
+Say your customer data is stored in the "customer" table while the "invoice"
+data is stored in it's own table. And invoice rows also have there own table.
+
+Searching (using the FilterQuery syntax) can be as simple as:
+`invoice_price: >"$20.00" invoice_row: ~*"my cool product"; customer_type: !consumer.`
```
Comment (cordoval) [inline README.md]: this example makes me understand what could this be used for, i think more of these examples could be good.

do you have custom hard coded queries like that? or is this more for a dashboard admin to be able to create their own user facing query builder so they can sift data through it? I have seen this done in jira for example to sift tickets. Is this the idea? if so you should even put this example and what other usages it maybe have.

File: README.md (hunk 3 above)
Comment (sstok) [inline README.md]: Yes, I'm talking about relation tables here. Invoice rows/details :) being a bit technical here.

File: README.md (hunk 4 above)
Comment (sstok) [inline README.md]: Good point :+1:

File: README.md (hunk 5 above)
Comment (sstok) [inline README.md]: Yes thats one use-case, it abstracts the query building into a search condition which can be used for any back-end storage like SQL or ElasticSearch. Thanks for helping with this :+1:

File: README.md [hunk 6]
```diff
@@ -10,30 +9,49 @@ README
...
+
+You just searched in three relational tables! using a single condition with a
```
Comment (cordoval) [inline README.md]: Using

File: README.md [hunk 7]
```diff
@@ -10,30 +9,49 @@ README
...
+user-friendly syntax. And that is just the start, checkout all the cool and
+powerful features RollerworksSearch has to offer you.
```
Comment (cordoval) [inline README.md]: - you

**Pull request closed without merging**